crates/topos-core/src/types.rs

```diff
@@ -0,0 +1,76 @@
+use topos_uci::{Certificate, CertificateId};
+
+use serde::{Deserialize, Serialize};
+
+use self::stream::{CertificateSourceStreamPosition, Position};
+use topos_api::grpc::{
+    checkpoints::SourceStreamPosition,
+    tce::v1::{ProofOfDelivery as GrpcProofOfDelivery, SignedReady},
+};
+
+pub mod stream;
+
+pub type Ready = String;
+pub type Signature = String;
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
+pub struct CertificateDelivered {
+    pub certificate: Certificate,
+    pub proof_of_delivery: ProofOfDelivery,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
+pub struct ProofOfDelivery {
+    pub certificate_id: CertificateId,
+    pub delivery_position: CertificateSourceStreamPosition,
+    pub readies: Vec<(Ready, Signature)>,
```

Comment:
What does readies means?

Reply:
it's the list of the `Ready` messages with their associated `Signature` received during the double echo

Reply:
Add comment to code :)

Reply:
While "readies" made me giggle, I'd consider renaming to `ready` to be more correct.

Reply:
We were thinking the same with @hadjiszs

How would you differentiate a `Ready` from a `vec` of `Ready` then?

Reply:
Hopefully context helps enough for it to be obvious if it's a collection or a single item. Otherwise we'd need a quirky name, `ready_sigs` or some such awkwardness. FWIW I personally don't mind having a name that is both crystal clear and also makes readers giggle a bit.